clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -1293,6 +1293,10 @@ compileModuleImpl(CompilerInstance &ImportingInstance, SourceLocation ImportLoc,
                                             diag::remark_module_build_done)
     << ModuleName;
 
+  // Propagate the statistics to the parent FileManager.
+  if (FrontendOpts.ModulesShareFileManager)
```

Comment:
Shouldn't this condition be inverted? When `FrontendOpts.ModulesShareFileManager == true`, I'd expect `&ImportingInstance.getFileManager() == &Instance.getFileManager()`. Maybe we can have an assertion in `FileManager::AddStats()` that would catch this.

Reply:
Yes, you're right, fixed!